test/spec/Feature/Query/RpcSpec.hs

```diff
@@ -1490,3 +1490,25 @@ spec actualPgVersion =
            `shouldRespondWith`
              [json| {"code":"22026","details":null,"hint":null,"message":"bit string length 6 does not match type bit(5)"} |]
            { matchStatus = 400 }
+
+    context "get message and details from raise sqlstate" $ do
+      it "gets message and details from raise sqlstate PGRST" $ do
+        get "/rpc/raise_sqlstate_test1" `shouldRespondWith`
+          [json|{"code":"123","message":"ABC","details":"DEF","hint":"XYZ"}|]
+          { matchStatus = 332
+          , matchHeaders = ["X-Header" <:> "str"] }
+
+        get "/rpc/raise_sqlstate_test2" `shouldRespondWith`
+          [json|{"code":"123","message":"ABC","details":null,"hint":null}|]
+          { matchStatus = 332
+          , matchHeaders = ["X-Header" <:> "str"] }
+
+      it "returns JSONParseError for invalid JSON in RAISE Message field" $
+        get "/rpc/raise_sqlstate_invalid_json_message" `shouldRespondWith`
+          [json|{"code":"PGRSTJ00","message":"The message and detail field of RAISE error expects JSON","details":null,"hint":null}|]
+          { matchStatus = 400 }
+
+      it "returns JSONParseError for invalid JSON in RAISE Details field" $
+        get "/rpc/raise_sqlstate_invalid_json_details" `shouldRespondWith`
+          [json|{"code":"PGRSTJ00","message":"The message and detail field of RAISE error expects JSON","details":null,"hint":null}|]
+          { matchStatus = 400 }
```

Comment:
@taimoorzaeem The end result looks good :+1:

As you mention, some refactoring is needed. Ideally we don't introduce attoparsec and stick with parsec.

Reply:
@steve-chavez For parsec, I would have to write a complete JSON parser from scratch which seems rather unnecessary.